src/Sulu/Bundle/AdminBundle/Resources/js/components/DatePicker/tests/DatePicker.test.js

```diff
@@ -13,52 +14,125 @@ beforeEach(() => {
 
 test('DatePicker should render', () => {
     const onChange = jest.fn();
-    expect(mount(<DatePicker onChange={onChange} value={null} />)).toMatchSnapshot();
+    const datePicker = mount(<DatePicker onChange={onChange} value={null} />);
+
+    expect(datePicker.render()).toMatchSnapshot();
+    expect(pretty(document.body ? document.body.innerHTML : '')).toMatchSnapshot();
+
+    datePicker.unmount();
```

Comment:
to keep tests/snapshots independent, we need to unmount the component (or use another way to clear the dom) after each test to remove the popover from the document body. unfortunatelly, calling `component.unmount()` at the end of every test is quite error prone, because it can be forgotten easily.

do we have a similar situation with another component? have we already implemented a less error prone solution there?

Reply:
We have a similar problem e.g. with the `DatagridStore`, which should have its `destroy` method called, because otherwise too many `autorun`s are lying around... Don't know about any better solution at the moment :-/

Reply:
But maybe it would work to get rid of `document.body` in these tests. I think if you query the `Portal` with enzyme, you can get whatever is rendered in there, without being dependant on `document.body`... Would at least be worth a try.